packages/server/src/api/report-schema/report-schema.controller.ts

```diff
@@ -106,6 +74,18 @@ class ReportSchemaController {
     res.json(updatedReport);
   }
 
+  private static async findReport(req: ApiRequest<OrgReportParams>) {
+    const id = req.params.reportId;
+    const org = req.appOrg!.id;
+    const report = await ReportSchema.findOne({ relations: ['org'], where: { id, org } });
+
+    if (!report) {
+      Log.error(`report_schema table is missing a record for ID ${id} and org ID ${org} `);
+      throw new NotFoundError('Report could not be found.');
+    }
```

Comment:
I think this function would be a little more flexible if we made a custom error type to throw (ie. `ReportSchemaNotFound(reportId, orgId)`) and didn't do the error check inside the `findReport` function, leaving the caller responsible for it instead. It would be more semantically in-line with the other database `find` operations that way, since none of those throw errors if nothing is found.

Reply:
This refactor effort started by a need of being able to quickly tell what is happening and what is needed when the app log shows: Error: Report could not be found.

I quickly noticed that this error is logged in 3 places and noticed the code duplication so instead of improving all 3 places I quickly extracted the coed out.

All you suggestions are great, no doubt but at some point I need to go back to my assigned work as I feel I am constantly behind on this project, not to mention other work I have to do for PE. 

This is a great suggestion but I would suggest that you create a tech debt task and this task is prioritized.